### PR TITLE
feat: support for hypercore

### PR DESCRIPTION
**Pull request by frolvanya**
## Summary

Adds a new `hypercore-bridge-client` crate plus connector/CLI wiring so a HyperCore user can call **one** Hyperliquid Core action (`sendToEvmWithData`) and have their tokens delivered to NEAR / Solana / any supported chain — no manual HyperEVM tx in between.

Two flows are exposed:
- **`ACTION_INIT_TRANSFER` (`0x01`)** — Core → NEAR/other-chain via `OmniBridge.initTransfer`. The HlBridgeToken contract releases the bridged amount from its system-address pool and routes it through the bridge.
- **`ACTION_TRANSFER` (`0x00`)** — Core → arbitrary HyperEVM recipient (pool release without bridge involvement).

Also corrects the HyperEVM bridge factory default address (`0x0000…` placeholder → `0xf353b40fC144d1c6c5BCdda712fa6De833016aF9`), matching the deployed proxy on both mainnet and testnet.

## Why

`HlBridgeToken` (omni-bridge PR #624) already accepts a `coreReceiveWithData(...)` callback from Hyperliquid's system address. What was missing was the SDK side: signing Hyperliquid user-signed actions (EIP-712), posting them to `/exchange`, and reliably correlating the resulting on-chain effect. This PR fills that gap.

## What's in the diff

**New crate: `bridge-sdk/bridge-clients/hypercore-bridge-client/`** (~890 LoC)

- `action.rs` — `SendToEvmWithDataAction` JSON wire struct, `HyperliquidNetwork` enum (mainnet/testnet URLs + chain ids 999/998), `format_amount(u128, decimals) → String` for the decimal-string wire format.
- `encoders.rs` — `encode_init_transfer_action(fee, &OmniAddress, &str)` and `encode_transfer_action(Address)`. Tag constants exposed.
- `signing.rs` — EIP-712 `sign_action()` with manual domain/struct hash + `keccak256(0x19 || 0x01 || dom || struct)`. Round-trip signer-recovery test included.
- `client.rs` (= `hypercore_bridge_client.rs`) — `HyperCoreBridgeClient`: posts the signed action to `/exchange`, then polls HyperEVM for the resulting `CoreReceived` log (filtered by `address + topic0 + topic1=signer + data == expected`). Returns the HyperEVM `TxHash` (or the full `CoreReceivedLog` via the `_detailed` variant).
- `builder.rs` — builder pattern matching the rest of the workspace, with `http_timeout` (default 30s), `poll_interval` (1.5s), `poll_timeout` (60s) all configurable.
- `error.rs` — `thiserror`-based errors flow through to `BridgeSdkError` via `From` impl in `bridge-connector-common`.

**Edits to existing crates**

- `evm-bridge-client`: new `sol! HlBridgeToken { event CoreReceived(...) }`, `CoreReceivedFilter`, `CoreInitiatedTransfer`, plus `get_core_received_log()` and `parse_core_initiated_transfer()` helpers. The correlator is needed because `OmniBridge.InitTransfer.sender` is the HlBridgeToken contract address (not the Core user) when the transfer was triggered via the callback — without the `CoreReceived` correlation, indexers can't attribute back to the originating Core account.
- `bridge-connector-common`: `From<HyperCoreBridgeClientError>` for `BridgeSdkError`, plus three new variants (`HyperCoreError`, `HyperCoreActionRejected`, `HyperCorePollTimeout`).
- `omni-connector`: new field `hypercore_bridge_client`, two new `InitTransferArgs` variants (`HyperCoreInitTransfer`, `HyperCoreEvmTransfer`), dispatch arms in `init_transfer`, `hypercore_bridge_client()` accessor. Doc-comment on `FinTransferArgs::EvmFinTransfer` explaining the contract-side message-length dispatch that delivers inbound supply to a HyperCore user.
- `bridge-cli`: `hypercore_api` + `hypercore_signature_chain_id` config fields wired through `or()` / `env_config()` / `default_config()` for mainnet/testnet/devnet; two subcommands `hyper-core-init-transfer` and `hyper-core-evm-transfer`. HyperEVM bridge factory address corrected from placeholder to deployed proxy.

## Verification

```
cargo build --workspace        →  PASS
cargo clippy --workspace       →  3 warnings, all pre-existing on main (verified via stash)
cargo test --workspace         →  PASS
cargo fmt --check              →  clean
cargo test -p hypercore-bridge-client  →  8/8 pass
```

The 8 unit tests cover:
- `format_amount` edge cases (zero, leading zeros, trailing-zero trim)
- ABI-encode/decode round trip for both `ACTION_TRANSFER` and `ACTION_INIT_TRANSFER` data layouts
- `signatureChainId` parsing
- EIP-712 domain separator determinism
- Struct hash changes with each field (sensitivity check)
- `sign_action_recovers_signer` — sign with a known key, recover from the digest, assert the recovered address matches

## Known risks / outstanding

✅ **EIP-712 type string validated against Hyperliquid testnet.** `cargo test -p hypercore-bridge-client testnet_l1 -- --ignored --nocapture` with two fresh random keys returned `{"status":"err","response":"Must deposit before performing actions. User: 0x<our-signing-address>"}` — Hyperliquid's L1 recovered exactly the address we signed with from our digest. Confirmed correct: field order, integer widths (`uint32 destinationChainId`, `uint64 gasLimit`, `uint64 nonce`), `signatureChainId` placement (domain only), inclusion of `sourceDex`/`addressEncoding` in the signed payload, and `data` serialized as `0x`-prefixed lowercase hex. Test stays as `#[ignore]` so CI doesn't hit network; doc comments in `signing.rs` reflect verified status. Don't use the Anvil `0xac09…` well-known key for the smoke test — that address is configured as a multi-sig account on testnet by someone else, which masks the address-recovery signal with a less informative `"Multi-sig required"` error.

⚠️ **No full money-transfer E2E in this PR.** The signing pipeline is verified, but an actual successful bridge transfer (funded HyperCore testnet account → linked spot token → deployed `HlBridgeToken` → bridge through to a NEAR account) hasn't been executed. Worth doing before any mainnet usage to confirm gas defaults, polling timeouts, and the `parse_core_initiated_transfer` correlator behave on real traffic. Less critical than the EIP-712 risk was — protocol-level shape is no longer the suspect.

ℹ️ `ChainKind::HyperCore` was **deferred** (Decision A). Inbound-to-Core works today via the contract-side message-length dispatch on `EvmFinTransfer { chain_kind: HyperEvm, … }` with a non-empty `message`; documented on the variant. Adding a dedicated `ChainKind::HyperCore` is a follow-up if the UX warrants it (requires an `omni-types` bump).

ℹ️ `ClaimFeeArgs` has no `WithVaaProofTx` variant — this affects all Wormhole-using chains (Arb, Base, Bnb, Pol, HyperEvm) equally, not specifically HyperEvm. Pre-existing gap.

## Test plan

- [ ] `cargo build --workspace` (already passes locally)
- [ ] `cargo test --workspace` (already passes locally)
- [ ] **E2E testnet smoke test** (gate before merge):
  - Deploy or use an existing `HlBridgeToken` on HyperEVM testnet
  - Fund a Core account with the linked spot token
  - `bridge-cli testnet hyper-core-init-transfer --token <NAME:tokenId> --hl-token <addr> --amount <wei> --decimals <u8> --recipient near:alice.testnet --fee 0 --message ref=hypercore`
  - Confirm `/exchange` returns `{"status":"ok"}`
  - Confirm `CoreReceived(sender=<core_user>, action=1, …)` lands on HyperEVM
  - Confirm `OmniBridge.InitTransfer` is emitted in the same tx with `sender == HlBridgeToken address`
  - Assert `EvmBridgeClient::parse_core_initiated_transfer(tx_hash, hl_bridge_token)` returns both events correlated
- [ ] If `/exchange` rejects with a signature error, iterate on the EIP-712 type list per the risk note above

### CODE REVIEW

File: bridge-cli/src/omni_connector_command.rs [hunk 1]
```diff
@@ -469,6 +470,46 @@ pub enum OmniConnectorSubCommand {
...
+        message: Option<String>,
+        #[clap(long, help = "Override HyperEVM gas limit for the system call")]
+        gas_limit: Option<u64>,
+        #[command(flatten)]
+        config_cli: CliConfig,
+    },
+    #[clap(
+        about = "HyperCore -> HyperEVM via sendToEvmWithData (ACTION_TRANSFER): pool release to an EVM recipient"
+    )]
+    HyperCoreEvmTransfer {
```
Comment (karim-en) [inline bridge-cli/src/omni_connector_command.rs]: Let's combine HyperCoreInitTransfer and HyperCoreEvmTransfer into one command.
Also later it will be better to combine all init tansfers commands into one

**Review by karim-en — CHANGES_REQUESTED** [review]

File: bridge-sdk/connectors/omni-connector/src/omni_connector.rs [hunk 2]
```diff
@@ -2749,9 +2800,92 @@ impl OmniConnector {
...
+                .hypercore_evm_transfer(
+                    token,
+                    hl_bridge_token,
+                    amount,
+                    decimals,
+                    evm_recipient,
+                    gas_limit,
+                )
+                .await
+                .map(|tx_hash| format!("{tx_hash:?}")),
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply bridge-sdk/connectors/omni-connector/src/omni_connector.rs]: Fixed in [f895223](https://github.com/Near-One/bridge-sdk-rs/pull/289/commits/f895223052e99e958aa2a3bd15e2cd5eab2d7007)

File: bridge-sdk/connectors/omni-connector/src/omni_connector.rs [hunk 3]
```diff
@@ -237,6 +241,43 @@ pub enum InitTransferArgs {
...
+    ///
+    /// - `token` is the Hyperliquid spot identifier (`"NAME:0x<32hex>"`), NOT
+    ///   the bridge ERC20 address.
+    /// - `amount` is in bridge ERC20 wei units. `decimals` is the bridge
+    ///   token's decimals. The connector formats `amount * 10^-decimals` as
+    ///   the Hyperliquid wire decimal string.
+    /// - The bridge token's decimals MUST equal Core's `szDecimals +
+    ///   evmExtraWeiDecimals` (the HyperCore<->HyperEVM linking invariant). If
+    ///   the token was linked incorrectly, the formatted amount will not
+    ///   match the user's Core spot balance precision.
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply bridge-sdk/connectors/omni-connector/src/omni_connector.rs]: Fixed in Fixed in [f895223](https://github.com/Near-One/bridge-sdk-rs/pull/289/commits/f895223052e99e958aa2a3bd15e2cd5eab2d7007)

File: bridge-sdk/bridge-clients/hypercore-bridge-client/src/hypercore_bridge_client.rs [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+            .get_block_number()
+            .await
+            .map_err(|e| HyperCoreBridgeClientError::Rpc(e.to_string()))?;
+        // Look back ~10 blocks in case the system tx already landed during the
+        // POST round-trip; HyperEVM small blocks are 1s.
+        let mut from_block = head_at_start.saturating_sub(10);
+
+        let start = Instant::now();
+        loop {
+            tokio::time::sleep(self.poll_interval).await;
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply bridge-sdk/bridge-clients/hypercore-bridge-client/src/hypercore_bridge_client.rs]: Fixed in [f895223](https://github.com/Near-One/bridge-sdk-rs/pull/289/commits/f895223052e99e958aa2a3bd15e2cd5eab2d7007)

File: bridge-cli/src/omni_connector_command.rs (hunk 1 above)
Comment (frolvanya) [reply bridge-cli/src/omni_connector_command.rs]: Merged in 9217e76a9f7dbe49b3f7ea28b1380758609dac61
